zig env: add support for line based output

perillo opened the pull request:
Currently it is not very convenient to extract a single environment variable from `zig env`.
An example using the shell: `zig env | jq -r .lib_dir`.

Update print_env.zig so that, when args is not empty, print each environment variable associated with an argument.
An updated example: `zig env lib_dir`.

Unrecognized environment variables are ignored.

andrewrk reviewed (changes requested):
Thank you for this patch. I have decided to copy `go env` for this.

* remove the json output
* use key="value" syntax
* use the same data for both kinds of outputs so that there is no chance of bugs in this code.

perillo commented:
> Thank you for this patch. I have decided to copy `go env` for this.
> 
>     * remove the json output
> 
>     * use key="value" syntax
> 
>     * use the same data for both kinds of outputs so that there is no chance of bugs in this code.

I think json output should remain, enabled by the `--json` flag, since it is probably more convenient when parsing all the envars.

perillo commented:
I removed the json output and moved all the environment variables in a single variable.

andrewrk commented:
I have rebased your branch, added a commit on top, and set the PR to auto-merge on completion of the CI tests. Example output of `zig env` now looks like this:

```
$ stage4/bin/zig env 
zig_exe=/home/andy/Downloads/zig/build-release/stage4/bin/zig
lib_dir=/home/andy/Downloads/zig/lib
std_dir=/home/andy/Downloads/zig/lib/std
global_cache_dir=/home/andy/.cache/zig
version=0.12.0-dev.1104+053119083
target=x86_64-linux.6.1.53...6.1.53-gnu.2.19
ZIG_GLOBAL_CACHE_DIR
ZIG_LOCAL_CACHE_DIR
ZIG_LIB_DIR
ZIG_LIBC
ZIG_BUILD_RUNNER
ZIG_VERBOSE_LINK
ZIG_VERBOSE_CC
ZIG_BTRFS_WORKAROUND
CC=gcc
NO_COLOR
XDG_CACHE_HOME
HOME=/home/andy
INCLUDE
```

squeek502 commented on src/resinator/preprocess.zig:
```diff
@@ -67,7 +68,7 @@ pub fn appendClangArgs(arena: Allocator, argv: *std.ArrayList([]const u8), optio
 
     if (!options.ignore_include_env_var) {
-        const INCLUDE = std.process.getEnvVarOwned(arena, "INCLUDE") catch "";
+        const INCLUDE = (introspect.EnvVar.INCLUDE.get(arena) catch @panic("OOM")) orelse "";
```
If this environment variable was guaranteed to only be read during `zig rc` and never any other `zig`  command, would this introspect code still be needed in the resinator code?

andrewrk replied:
No, I think if that guarantee was upheld, then the resinator patches here could be reverted :+1:

perillo commented:
> I have rebased your branch, added a commit on top, and set the PR to auto-merge on completion of the CI tests. Example output of `zig env` now looks like this:
>

Thanks for the merge.
 
> ```
> $ stage4/bin/zig env 
> zig_exe=/home/andy/Downloads/zig/build-release/stage4/bin/zig
> lib_dir=/home/andy/Downloads/zig/lib
> std_dir=/home/andy/Downloads/zig/lib/std
> global_cache_dir=/home/andy/.cache/zig
> version=0.12.0-dev.1104+053119083
> target=x86_64-linux.6.1.53...6.1.53-gnu.2.19
> ZIG_GLOBAL_CACHE_DIR
> ...
> ```

In `go env` an empty environment variable is rendered as
`ZIG_GLOBAL_CACHE_DIR=''`

Also, I think that showing both Zig **internal** variables and environment variables may be confusing.

andrewrk commented:
> In `go env` an empty environment variable is rendered as `ZIG_GLOBAL_CACHE_DIR=''`

I think it is better to not use quoted strings, relying on newline for the delimiter. With this way, one can easily parse or grep for `=` to determine set variables.

> Also, I think that showing both Zig **internal** variables and **environment** variables may be confusing.

I think it's fine. Do you have a real world use case to motivate further changes?
